### PR TITLE
[MTP] Refactor mtp predictor to avoid d2h operation

### PR DESCRIPTION
**Pull request by MengqingCao**
## Purpose
Refactor mtp predictor to avoid d2h operation.
The original logic brings device-to-host operation as `position` is a device tensor and the indexing operation `inputs_embeds[positions == 0]` will trigger a d2h synchornization. This breaks the static graph mode in vllm-ascend.
This pr refactors it to all the operations on torch, thus fixing this issue.

## Test Plan
Test locally on both H20 and vllm-ascend with deepseek-v3 + mtp + piecewise cudagraph

## Test Result

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

**Comment by MengqingCao** [comment]
Good day! @luccafong Could you take a look at this pr? thanks!

**Review by luccafong — APPROVED** [review]

File: vllm/model_executor/models/deepseek_mtp.py [hunk 1]
```diff
@@ -97,7 +97,7 @@ def forward(
     ) -> torch.Tensor:
         assert inputs_embeds is not None
         # masking inputs at position 0, as not needed by MTP
-        inputs_embeds[positions == 0] = 0
+        inputs_embeds = torch.where(positions.unsqueeze(-1) == 0, 0, inputs_embeds)
```
Comment (benchislett) [inline vllm/model_executor/models/deepseek_mtp.py:100]: does this issue the same number of kernels as `inputs_embeds.masked_fill_(positions == 0, 0)`? I would prefer to use `masked_fill_` if there is no performance loss, and I'm not sure if they are equivalent to torch

Comment (MengqingCao) [reply vllm/model_executor/models/deepseek_mtp.py:100]: @benchislett I got your point of using an in-place operation to avoid introduing memory allocating operation. And I checked the source code of `masked_fill_`, it indicates that `masked_fill` actually is implemented by `where` operation inside, and adding some checks before that. Thus I think it's fine to keep the `torch.where` here? WDYT?
https://github.com/pytorch/pytorch/blob/main/torch/_refs/__init__.py#L5892

Comment (benchislett) [reply vllm/model_executor/models/deepseek_mtp.py:100]: In that case it's fine either way.

**Review by benchislett — APPROVED** [review]
LGTM

**Comment by MengqingCao** [comment]
> LGTM

Thanks! could you help merge this pr? I also hope it will be included in releases/v0.11.1, should I make a backport pr then?

**Review by whx-sjtu — APPROVED** [review]
Nice fix!
